docs/copilot/guides/large-complex-codebases-guide.md

```diff
@@ -0,0 +1,408 @@
+- **Concept-level** (context-triggered): Cross-cutting concerns like dependency injection, logging
+
+Use the `applyTo` frontmatter property for file-pattern-based context:
+
+```markdown
+---
+applyTo: "src/frontend/**/*"  # Frontend team context
+---
+# Frontend Development Guidelines
+Reference [general coding standards](../copilot-instructions.md) for shared conventions.
+...
+```
+
+Use `description` only for context-triggered conceptual guidance:
+
+```markdown
+---
+description: "Enterprise logging patterns and structured logging best practices"
+---
+# Logging Standards
+Guidelines for implementing logging, monitoring, and observability features...
+```
+
+The AI agent can automatically load this instruction file when it detects the conversation involves logging, monitoring, or observability concepts.### Context scoping strategies
+
+Use `applyTo` patterns to limit AI context to relevant subsystems:
+
+- `.github/copilot-instructions.md` for repository-wide context
+- Focused `.instructions.md` files with specific `applyTo` patterns for subsystems
+- Reference documentation using Markdown links
+- Define module boundaries in module-specific instruction files
+
+Example scoping patterns:
+
+```markdown
+---
+applyTo: "src/api/**/*,tests/api/**/*"  # Backend API and tests only
+---
+
+---
+applyTo: "src/frontend/**/*.{ts,tsx}"   # Frontend TypeScript only
+---
+
+---
+applyTo: "**/*.{test,spec}.{js,ts}"     # Test files only
+---
+```
+
+## Advanced chat mode patterns
+
+### Role-based chat modes
+
+Different roles need different AI personas and tool access. Create [custom chat modes](/docs/copilot/customization/custom-chat-modes.md) with specific instructions and tool sets:
+
+```
+.github/chatmodes/
+  architect.chatmode.md      # System design, cross-cutting concerns
+  frontend-dev.chatmode.md   # UI components, user experience
+  backend-dev.chatmode.md    # APIs, data models, integrations
+  devops.chatmode.md         # Infrastructure, deployment, monitoring
+  security.chatmode.md       # Security review, threat modeling
+```
+
+Each chat mode file defines specific tools and instructions for that role's workflow.
+
+<!-- TODO: Add complete example chat mode files with tool configurations for each role -->
+
+### Team-specific workflows
+
+Create chat modes that encode team workflows:
+
+- **Research modes**: Read-only exploration using `search`, `codebase`, `usages` tools
+- **Migration modes**: Refactoring with specific instructions for technical debt reduction
+- **Integration modes**: Cross-service communication patterns with architectural context
+- **Review modes**: Code quality assessment with team guidelines
+
+### Cross-functional modes
+
+For work spanning multiple teams:
+
+- **API design mode**: Backend + frontend collaboration
+- **Performance mode**: Application + infrastructure optimization
+- **Migration mode**: Coordinated changes across multiple services
+
+## Context compaction techniques
+
```

Comment:
Quick intro about the problems related to context when working with a large codespace.